src/Compilers/CSharp/Portable/BoundTree/Statement.cs

```diff
@@ -16,6 +16,10 @@ internal partial class BoundStatement : IStatement
 
         SyntaxNode IOperation.Syntax => this.Syntax;
 
+        ITypeSymbol IOperation.Type => null;
```

Comment:
I'm not sure that the predefined types are available from within the implementation of bound trees. I'm also prepared to defend the distinction between "could have a type that turns out to be void" and "cannot have a type". A null type implies the latter.